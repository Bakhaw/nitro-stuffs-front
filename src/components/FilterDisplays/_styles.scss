.FilterDisplays {
  display: flex;
  flex-direction: row;
  margin-bottom: 15px;
  align-items: center;

  &__Image {
    opacity: 0.5;
    cursor: pointer;
    margin: 10px;
  }

  &__ImageActive {
    opacity: 1;
  }
}
