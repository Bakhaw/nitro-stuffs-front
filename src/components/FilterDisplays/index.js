import React from 'react';

import Image from '../Image';

function FilterDisplays({ isDefaultDisplay, handleFilterDisplaysClick }) {
  const displays = [
    {
      key: 'map',
      icon: 'miniMap',
      additionalStyle: `${
        isDefaultDisplay ? 'FilterDisplays__ImageActive' : ''
      }`,
      size: `${isDefaultDisplay ? '50' : '30'}`,
    },
    {
      key: 'video',
      icon: 'camera',
      additionalStyle: `${
        !isDefaultDisplay ? 'FilterDisplays__ImageActive' : ''
      }`,
      size: `${!isDefaultDisplay ? '50' : '30'}`,
    },
  ];
  return (
    <div className='FilterDisplays'>
      {displays.map((display) => (
        <div
          key={display.key}
          className={`FilterDisplays__Image ${display.additionalStyle}`}
          onClick={handleFilterDisplaysClick}
        >
          <Image
            width={display.size}
            height={display.size}
            iconType={display.icon}
          />
        </div>
      ))}
    </div>
  );
}

export default FilterDisplays;
