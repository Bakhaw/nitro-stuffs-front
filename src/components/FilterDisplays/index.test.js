import React from 'react';
import { shallow } from 'enzyme';

import FilterDisplays from '.';

test('should render properly', () => {
  const props = {
    isDefaultDisplay: true,
    handleFilterDisplaysClick: jest.fn(),
  };

  const container = shallow(<FilterDisplays {...props} />);
  expect(container.debug()).toMatchSnapshot();
});
