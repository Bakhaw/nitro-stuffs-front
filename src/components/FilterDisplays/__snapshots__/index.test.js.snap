// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
"<div className=\\"FilterDisplays\\">
  <div className=\\"FilterDisplays__Image FilterDisplays__ImageActive\\" onClick={[Function: mockConstructor]}>
    <Image width=\\"50\\" height=\\"50\\" iconType=\\"miniMap\\" />
  </div>
  <div className=\\"FilterDisplays__Image \\" onClick={[Function: mockConstructor]}>
    <Image width=\\"30\\" height=\\"30\\" iconType=\\"camera\\" />
  </div>
</div>"
`;
